06_symbology_functions.sql

```diff
@@ -376,6 +376,31 @@ BEGIN
 END; $BODY$
 LANGUAGE plpgsql VOLATILE;
 
+
+--------------------------------------------------
+-- ON WASTEWATER NETWORKELEMENT DELETE
+--------------------------------------------------
+
+CREATE OR REPLACE FUNCTION qgep_od.on_wastewater_networkelement_delete()
+  RETURNS trigger AS
+$BODY$
+DECLARE
+  channel_id text;
+  reach_count integer;
+BEGIN
+  -- delete channel if no reach left
+  SELECT COUNT(fk_wastewater_structure) INTO reach_count
+    FROM qgep_od.wastewater_networkelement
+    WHERE fk_wastewater_structure = OLD.fk_wastewater_structure;
```

Comment:
This will also delete wastewater structures (manholes etc.) when the last wastewater node is removed, no?

Reply:
Add a warning if this is happening!

Reply:
I don't think so, let's try to write it down:

1. Trigger is on delete on a network element 
2. It counts the number of network elements having a reference to the same wastewater structure (i.e. channel)
2. Deletes channel and wastewater structure having obj_id as the channel one.

On 1., shall I check that it's indeed a reach (and not a reach point)?

Reply:
The proposed check sounds good.
So just for the record: a networkelement can be a reach or a wastewater node.